components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/factory/UpdateRequestFactory.java

```diff
@@ -0,0 +1,146 @@
+
+  private static Log log = LogFactory.getLog(UpdateRequestFactory.class);
+
+  @Override
+  public boolean canHandle(HttpServletRequest request, HttpServletResponse response) {
+
+    boolean canHandle = false;
+    if (request != null) {
+      Matcher matcher = DCRConstants.DCRM_ENDPOINT_CLIENT_CONFIGURATION_URL_PATTERN
+          .matcher(request.getRequestURI());
+      if (matcher.matches() && HttpMethod.PUT.equals(request.getMethod())) {
+        canHandle = true;
+      }
+    }
+    if (log.isDebugEnabled()) {
+      log.debug("canHandle " + canHandle + " by UpdateRequestFactory.");
+    }
+    return canHandle;
+  }
+
+  @Override
+  public UpdateRequest.UpdateRequestBuilder create(HttpServletRequest request,
+      HttpServletResponse response) throws FrameworkClientException {
+
+    UpdateRequest.UpdateRequestBuilder updateRequestBuilder = new UpdateRequest
+        .UpdateRequestBuilder(request, response);
+    create(updateRequestBuilder, request, response);
+    return updateRequestBuilder;
+  }
+
+  @Override
+  public void create(IdentityRequest.IdentityRequestBuilder builder, HttpServletRequest request,
+                      HttpServletResponse response) throws FrameworkClientException {
+
+    UpdateRequest.UpdateRequestBuilder updateRequestBuilder =
+        (UpdateRequest.UpdateRequestBuilder) builder;
+
+    super.create(updateRequestBuilder, request, response);
+
+    try {
+      Reader requestBodyReader = request.getReader();
+      JSONParser jsonParser = new JSONParser();
+      JSONObject jsonData = (JSONObject) jsonParser.parse(requestBodyReader);
+      if (log.isDebugEnabled()) {
+        log.debug("DCR request json : " + jsonData.toJSONString());
+      }
+
+      UpdateRequestProfile updateRequestProfile = updateRequestBuilder.getUpdateRequestProfile();
+
+      if (updateRequestProfile == null) {
+        updateRequestProfile = new UpdateRequestProfile();
+      }
+
+      String consumerKey = null;
+      Matcher matcher = DCRConstants.DCRM_ENDPOINT_CLIENT_CONFIGURATION_URL_PATTERN
+          .matcher(request.getRequestURI());
+      if (matcher.find()) {
+        consumerKey = matcher.group(2);
+      }
+
+      updateRequestProfile.setConsumerKey(consumerKey);
+
+      updateRequestProfile.setClientId((String) jsonData.get(UpdateRequestConstant.CLIENT_ID));
+      updateRequestProfile.setClientSecret(
+          (String) jsonData.get(UpdateRequestConstant.CLIENT_SECRET));
+      updateRequestProfile.setClientName((String) jsonData.get(UpdateRequestConstant.CLIENT_NAME));
+
+      Object object = jsonData.get(UpdateRequestConstant.REDIRECT_URIS);
+      if (object !=null && object instanceof JSONArray) {
+        JSONArray redirectUris = (JSONArray) object;
+        for (Object redirectUri : redirectUris) {
+          updateRequestProfile.getRedirectUris().add(redirectUri.toString());
+        }
+      } else if (object != null) {
+        updateRequestProfile.getRedirectUris().add((String) object);
```

Comment:
What happens if the user sends something other than a string for the redirect uri?